Ninja.cmake

```diff
@@ -47,9 +48,8 @@ else()
 endif()
 
 find_program(NINJA_PATH
```

Comment:
My misguided use of `find_program` was an attempt to avoid the HINTS path in the case when a consumer wants to specify the exact version to use. The search order specified in [the `find_program` documentation](https://cmake.org/cmake/help/latest/command/find_program.html) says that the HINTS path is searched fourth, after cache variables and environment variables. So even if a user were to specify a, say, `NINJA_DIR` it wouldn't guarantee that that version would be used, it would vary based on the user's environment. Which seems non-deterministic - if a user is specifying a variable to find a program, then the explicit action should be respected.

So the 'cheapest' fix - IMHO - would be to remove the 'PATHS' parameter on the original, i.e.:

```cmake
find_program(NINJA_PATH
    NAMES ninja ninja.exe
)
```

`find_program` doesn't appear to update `NINJA_PATH` if it's already set - allowing a user to specify an explicit path - but then falls back to documented behavior if `NINJA_PATH` isn't set. Adding `NINJA_DIR` just adds more heuristics to the search, which doesn't seem necessary.

Reply:
ok